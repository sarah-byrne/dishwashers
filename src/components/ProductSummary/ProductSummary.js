import React from "react";

const ProductSummary = props => (
  <div id={`dishwasherSummary${props.index}`}>
    <img id={`image${props.index}`} src={props.product.image} />
    <div id={`title${props.index}`}>{props.product.title}</div>
    <div id={`price${props.index}`}>Â£{props.product.price.now}</div>
  </div>
);

export default ProductSummary;
