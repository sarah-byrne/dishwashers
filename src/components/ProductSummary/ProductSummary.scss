@import "node_modules/jl-ui-kit/src/_jlUiKit.scss";

.summary {
  @include display-as($body);
  position: relative;
  width: 100%;
  height: 100%;
}
