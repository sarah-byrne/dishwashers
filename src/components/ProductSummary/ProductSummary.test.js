import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import ProductSummary from "./ProductSummary";

describe("ProductSummary", () => {
  const product = {
    price: { now: "10" },
    image: "//johnlewis.scene7.com/is/image/JohnLewis/233326789?",
    title: "Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White"
  };
  const summary = shallow(
    <ProductSummary product={product} key={0} index={0} />
  );

  it("displays the product summary", () => {
    expect(summary).toMatchSnapshot();
  });

  it("displays the price", () => {
    const price = "10";
    const prodPrice = summary.find("#price0");

    expect(prodPrice.text()).toEqual(`Â£${price}`);
  });

  it("displays the title", () => {
    const title = "Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White";
    const prodTitle = summary.find("#title0");

    expect(prodTitle.text()).toEqual(`${title}`);
  });

  it("displays the image", () => {
    const picture = summary.find("#image0");

    expect(picture.prop("src")).toEqual(
      "//johnlewis.scene7.com/is/image/JohnLewis/233326789?"
    );
  });
});
