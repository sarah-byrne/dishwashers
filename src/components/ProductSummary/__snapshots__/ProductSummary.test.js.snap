// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSummary displays the product summary 1`] = `
<div
  id="dishwasherSummary0"
>
  <img
    id="image0"
    src="//johnlewis.scene7.com/is/image/JohnLewis/233326789?"
  />
  <div
    id="title0"
  >
    Indesit DIF 04B1 Ecotime Fully Integrated Dishwasher, White
  </div>
  <div
    id="price0"
  >
    Â£
    10
  </div>
</div>
`;
